src/explore-education-statistics-admin/src/pages/publication/components/PublicationAddExistingUsersTab.tsx

```diff
@@ -0,0 +1,104 @@
+import WarningMessage from '@common/components/WarningMessage';
+import React from 'react';
+import releasePermissionService, {
+  ManageAccessPageContributor,
+} from '@admin/services/releasePermissionService';
+import { Formik } from 'formik';
+import { Form, FormFieldCheckboxGroup } from '@common/components/form';
+import Button from '@common/components/Button';
+import useFormSubmit from '@common/hooks/useFormSubmit';
+import { generatePath, useHistory } from 'react-router-dom';
+import { publicationManageTeamAccessReleaseRoute } from '@admin/routes/routes';
+import { ReleaseRouteParams } from '@admin/routes/releaseRoutes';
+import { Release } from '@admin/services/releaseService';
+
+interface AddExistingUsersFormValues {
+  users: string[];
+}
+
+export interface Props {
+  publicationId: string;
+  release: Release;
+  contributors: ManageAccessPageContributor[];
+}
+
+const PublicationAddExistingUsersTab = ({
+  publicationId,
+  release,
+  contributors,
+}: Props) => {
+  const history = useHistory();
+
+  const handleSubmit = useFormSubmit<AddExistingUsersFormValues>(
+    async values => {
+      await releasePermissionService.updateReleaseContributors(
+        release.id,
+        values.users,
+      );
+      history.push(
+        generatePath<ReleaseRouteParams>(
+          publicationManageTeamAccessReleaseRoute.path,
+          {
+            publicationId,
+            releaseId: release.id,
+          },
+        ),
+      );
+    },
+    [],
+  );
+
+  if (!contributors || !contributors.length) {
+    return (
+      <WarningMessage>
+        There are no contributors for this release's publication.
+      </WarningMessage>
+    );
+  }
+
+  const initialValues: AddExistingUsersFormValues = {
+    users: contributors
+      .filter((c: ManageAccessPageContributor) => c.releaseRoleId !== undefined)
```

Comment:
I don't think we need to specify the type here, Typescript should be able to infer the type